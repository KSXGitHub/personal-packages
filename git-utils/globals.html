<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>@khai96x/git-utils — Reference</title>
	<meta name="description" content="Documentation for @khai96x/git-utils — Reference">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="assets/css/main.css">
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="assets/js/search.json" data-base=".">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="index.html" class="title">@khai96x/git-utils — Reference</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="globals.html">Globals</a>
				</li>
			</ul>
			<h1>@khai96x/git-utils — Reference</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<section class="tsd-panel-group tsd-index-group">
				<h2>Index</h2>
				<section class="tsd-panel tsd-index-panel">
					<div class="tsd-index-content">
						<section class="tsd-index-section tsd-is-external">
							<h3>Interfaces</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-interface tsd-is-external"><a href="interfaces/chunk.html" class="tsd-kind-icon">Chunk</a></li>
								<li class="tsd-kind-interface tsd-is-external"><a href="interfaces/gitstatus.html" class="tsd-kind-icon">Git<wbr>Status</a></li>
								<li class="tsd-kind-interface tsd-is-external"><a href="interfaces/param.html" class="tsd-kind-icon">Param</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Variables</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-variable tsd-is-external"><a href="globals.html#porcelain_status_regex" class="tsd-kind-icon">PORCELAIN_<wbr>STATUS_<wbr>REGEX</a></li>
								<li class="tsd-kind-variable"><a href="globals.html#setaur" class="tsd-kind-icon">set<wbr>Aur</a></li>
							</ul>
						</section>
						<section class="tsd-index-section tsd-is-external">
							<h3>Functions</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-function tsd-is-external"><a href="globals.html#findreporoot" class="tsd-kind-icon">find<wbr>Repo<wbr>Root</a></li>
								<li class="tsd-kind-function tsd-is-external"><a href="globals.html#gitlsfiles" class="tsd-kind-icon">git<wbr>LsFiles</a></li>
								<li class="tsd-kind-function tsd-is-external"><a href="globals.html#gitstatus" class="tsd-kind-icon">git<wbr>Status</a></li>
								<li class="tsd-kind-function tsd-is-external"><a href="globals.html#lines" class="tsd-kind-icon">lines</a></li>
								<li class="tsd-kind-function tsd-is-external"><a href="globals.html#main" class="tsd-kind-icon">main</a></li>
								<li class="tsd-kind-function tsd-is-external"><a href="globals.html#parseporcelainline" class="tsd-kind-icon">parse<wbr>Porcelain<wbr>Line</a></li>
								<li class="tsd-kind-function tsd-is-external"><a href="globals.html#program" class="tsd-kind-icon">program</a></li>
								<li class="tsd-kind-function tsd-is-external"><a href="globals.html#untrackedfiles" class="tsd-kind-icon">untracked<wbr>Files</a></li>
							</ul>
						</section>
						<section class="tsd-index-section tsd-is-external">
							<h3>Object literals</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-object-literal tsd-is-external"><a href="globals.html#spawnoptions" class="tsd-kind-icon">spawn<wbr>Options</a></li>
							</ul>
						</section>
					</div>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Variables</h2>
				<section class="tsd-panel tsd-member tsd-kind-variable tsd-is-external">
					<a name="porcelain_status_regex" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> PORCELAIN_<wbr>STATUS_<wbr>REGEX</h3>
					<div class="tsd-signature tsd-kind-icon">PORCELAIN_<wbr>STATUS_<wbr>REGEX<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">RegExp</span><span class="tsd-signature-symbol"> = /^(?&lt;status&gt;..) (?&lt;name&gt;.*)$/</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/KSXGitHub/personal-packages/blob/cd24d83/packages/git-utils/lib/browse-git-files.ts#L34">lib/browse-git-files.ts:34</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-variable">
					<a name="setaur" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> set<wbr>Aur</h3>
					<div class="tsd-signature tsd-kind-icon">set<wbr>Aur<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">string</span><span class="tsd-signature-symbol"> = require.resolve(&#x27;./bin/git-set-aur&#x27;)</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/KSXGitHub/personal-packages/blob/cd24d83/packages/git-utils/index.ts#L1">index.ts:1</a></li>
						</ul>
					</aside>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group tsd-is-external">
				<h2>Functions</h2>
				<section class="tsd-panel tsd-member tsd-kind-function tsd-is-external">
					<a name="findreporoot" class="tsd-anchor"></a>
					<h3>find<wbr>Repo<wbr>Root</h3>
					<ul class="tsd-signatures tsd-kind-function tsd-is-external">
						<li class="tsd-signature tsd-kind-icon">find<wbr>Repo<wbr>Root<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/KSXGitHub/personal-packages/blob/cd24d83/packages/git-utils/lib/browse-git-files.ts#L52">lib/browse-git-files.ts:52</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function tsd-is-external">
					<a name="gitlsfiles" class="tsd-anchor"></a>
					<h3>git<wbr>LsFiles</h3>
					<ul class="tsd-signatures tsd-kind-function tsd-is-external">
						<li class="tsd-signature tsd-kind-icon">git<wbr>LsFiles<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">AsyncGenerator</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/KSXGitHub/personal-packages/blob/cd24d83/packages/git-utils/lib/browse-git-files.ts#L24">lib/browse-git-files.ts:24</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">AsyncGenerator</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function tsd-is-external">
					<a name="gitstatus" class="tsd-anchor"></a>
					<h3>git<wbr>Status</h3>
					<ul class="tsd-signatures tsd-kind-function tsd-is-external">
						<li class="tsd-signature tsd-kind-icon">git<wbr>Status<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">AsyncGenerator</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/KSXGitHub/personal-packages/blob/cd24d83/packages/git-utils/lib/browse-git-files.ts#L42">lib/browse-git-files.ts:42</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">AsyncGenerator</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function tsd-is-external">
					<a name="lines" class="tsd-anchor"></a>
					<h3>lines</h3>
					<ul class="tsd-signatures tsd-kind-function tsd-is-external">
						<li class="tsd-signature tsd-kind-icon">lines<span class="tsd-signature-symbol">(</span>chunks<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">AsyncIterable</span><span class="tsd-signature-symbol">&lt;</span><a href="interfaces/chunk.html" class="tsd-signature-type">Chunk</a><span class="tsd-signature-symbol">&gt;</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">AsyncGenerator</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/KSXGitHub/personal-packages/blob/cd24d83/packages/git-utils/lib/browse-git-files.ts#L15">lib/browse-git-files.ts:15</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>chunks: <span class="tsd-signature-type">AsyncIterable</span><span class="tsd-signature-symbol">&lt;</span><a href="interfaces/chunk.html" class="tsd-signature-type">Chunk</a><span class="tsd-signature-symbol">&gt;</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">AsyncGenerator</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function tsd-is-external">
					<a name="main" class="tsd-anchor"></a>
					<h3>main</h3>
					<ul class="tsd-signatures tsd-kind-function tsd-is-external">
						<li class="tsd-signature tsd-kind-icon">main<span class="tsd-signature-symbol">(</span>param<span class="tsd-signature-symbol">: </span><a href="interfaces/param.html" class="tsd-signature-type">Param</a><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">number</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/KSXGitHub/personal-packages/blob/cd24d83/packages/git-utils/lib/browse-git-files.ts#L78">lib/browse-git-files.ts:78</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>param: <a href="interfaces/param.html" class="tsd-signature-type">Param</a></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">number</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function tsd-is-external">
					<a name="parseporcelainline" class="tsd-anchor"></a>
					<h3>parse<wbr>Porcelain<wbr>Line</h3>
					<ul class="tsd-signatures tsd-kind-function tsd-is-external">
						<li class="tsd-signature tsd-kind-icon">parse<wbr>Porcelain<wbr>Line<span class="tsd-signature-symbol">(</span>line<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="interfaces/gitstatus.html" class="tsd-signature-type">GitStatus</a><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">null</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/KSXGitHub/personal-packages/blob/cd24d83/packages/git-utils/lib/browse-git-files.ts#L36">lib/browse-git-files.ts:36</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>line: <span class="tsd-signature-type">string</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <a href="interfaces/gitstatus.html" class="tsd-signature-type">GitStatus</a><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">null</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function tsd-is-external">
					<a name="program" class="tsd-anchor"></a>
					<h3>program</h3>
					<ul class="tsd-signatures tsd-kind-function tsd-is-external">
						<li class="tsd-signature tsd-kind-icon">program<span class="tsd-signature-symbol">(</span>defaultFuzzyFinder<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">string</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">never</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/KSXGitHub/personal-packages/blob/cd24d83/packages/git-utils/lib/browse-git-files.ts#L104">lib/browse-git-files.ts:104</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5><span class="tsd-flag ts-flagOptional">Optional</span> defaultFuzzyFinder: <span class="tsd-signature-type">string</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">Promise</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">never</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-function tsd-is-external">
					<a name="untrackedfiles" class="tsd-anchor"></a>
					<h3>untracked<wbr>Files</h3>
					<ul class="tsd-signatures tsd-kind-function tsd-is-external">
						<li class="tsd-signature tsd-kind-icon">untracked<wbr>Files<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">AsyncGenerator</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">&gt;</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/KSXGitHub/personal-packages/blob/cd24d83/packages/git-utils/lib/browse-git-files.ts#L62">lib/browse-git-files.ts:62</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">AsyncGenerator</span><span class="tsd-signature-symbol">&lt;</span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">void</span><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">any</span><span class="tsd-signature-symbol">&gt;</span></h4>
						</li>
					</ul>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group tsd-is-external">
				<h2>Object literals</h2>
				<section class="tsd-panel tsd-member tsd-kind-object-literal tsd-is-external">
					<a name="spawnoptions" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagConst">Const</span> spawn<wbr>Options</h3>
					<div class="tsd-signature tsd-kind-icon">spawn<wbr>Options<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">object</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/KSXGitHub/personal-packages/blob/cd24d83/packages/git-utils/lib/browse-git-files.ts#L7">lib/browse-git-files.ts:7</a></li>
						</ul>
					</aside>
					<section class="tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-object-literal tsd-is-external">
						<a name="spawnoptions.stdio" class="tsd-anchor"></a>
						<h3>stdio</h3>
						<div class="tsd-signature tsd-kind-icon">stdio<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-symbol">(</span><span class="tsd-signature-type">"pipe"</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">"inherit"</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">[]</span><span class="tsd-signature-symbol"> = [&#x27;pipe&#x27;, &#x27;pipe&#x27;, &#x27;inherit&#x27;]</span></div>
						<aside class="tsd-sources">
							<ul>
								<li>Defined in <a href="https://github.com/KSXGitHub/personal-packages/blob/cd24d83/packages/git-utils/lib/browse-git-files.ts#L8">lib/browse-git-files.ts:8</a></li>
							</ul>
						</aside>
					</section>
				</section>
			</section>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class="globals current ">
						<a href="globals.html"><em>Globals</em></a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
					<li class=" tsd-kind-interface tsd-is-external">
						<a href="interfaces/chunk.html" class="tsd-kind-icon">Chunk</a>
					</li>
					<li class=" tsd-kind-interface tsd-is-external">
						<a href="interfaces/gitstatus.html" class="tsd-kind-icon">Git<wbr>Status</a>
					</li>
					<li class=" tsd-kind-interface tsd-is-external">
						<a href="interfaces/param.html" class="tsd-kind-icon">Param</a>
					</li>
					<li class=" tsd-kind-variable tsd-is-external">
						<a href="globals.html#porcelain_status_regex" class="tsd-kind-icon">PORCELAIN_<wbr>STATUS_<wbr>REGEX</a>
					</li>
					<li class=" tsd-kind-variable">
						<a href="globals.html#setaur" class="tsd-kind-icon">set<wbr>Aur</a>
					</li>
					<li class=" tsd-kind-function tsd-is-external">
						<a href="globals.html#findreporoot" class="tsd-kind-icon">find<wbr>Repo<wbr>Root</a>
					</li>
					<li class=" tsd-kind-function tsd-is-external">
						<a href="globals.html#gitlsfiles" class="tsd-kind-icon">git<wbr>LsFiles</a>
					</li>
					<li class=" tsd-kind-function tsd-is-external">
						<a href="globals.html#gitstatus" class="tsd-kind-icon">git<wbr>Status</a>
					</li>
					<li class=" tsd-kind-function tsd-is-external">
						<a href="globals.html#lines" class="tsd-kind-icon">lines</a>
					</li>
					<li class=" tsd-kind-function tsd-is-external">
						<a href="globals.html#main" class="tsd-kind-icon">main</a>
					</li>
					<li class=" tsd-kind-function tsd-is-external">
						<a href="globals.html#parseporcelainline" class="tsd-kind-icon">parse<wbr>Porcelain<wbr>Line</a>
					</li>
					<li class=" tsd-kind-function tsd-is-external">
						<a href="globals.html#program" class="tsd-kind-icon">program</a>
					</li>
					<li class=" tsd-kind-function tsd-is-external">
						<a href="globals.html#untrackedfiles" class="tsd-kind-icon">untracked<wbr>Files</a>
					</li>
					<li class=" tsd-kind-object-literal tsd-is-external">
						<a href="globals.html#spawnoptions" class="tsd-kind-icon">spawn<wbr>Options</a>
					</li>
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="assets/js/main.js"></script>
</body>
</html>