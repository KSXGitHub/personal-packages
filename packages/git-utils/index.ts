export const setAur = require.resolve('./bin/git-set-aur')
export * from './lib/browse-git-files'
export * from './lib/parse-porcelain-status'
