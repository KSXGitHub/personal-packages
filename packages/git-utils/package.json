{
  "name": "@khai96x/git-utils",
  "version": "0.0.12",
  "description": "Convenient git subcommands",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/KSXGitHub/personal-packages#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/KSXGitHub/personal-packages.git"
  },
  "bugs": {
    "url": "https://github.com/KSXGitHub/personal-packages/issues"
  },
  "keywords": [
    "git",
    "util"
  ],
  "main": "index.js",
  "module": "index.mjs",
  "browser": "index.mjs",
  "bin": {
    "porcelain-git-status-to-json": "./bin/porcelain-git-status-to-json",
    "browse-git-files": "./bin/browse-git-files",
    "git-browse-files": "./bin/browse-git-files",
    "git-fuzzy-add": "./bin/git-fuzzy-add",
    "git-fuzzy-reset": "./bin/git-fuzzy-reset",
    "git-fuzzy-restore": "./bin/git-fuzzy-restore",
    "git-fuzzy-delete": "./bin/git-fuzzy-delete",
    "git-sk": "./bin/git-sk",
    "git-fzf": "./bin/git-fzf",
    "git-set-aur": "./bin/git-set-aur"
  },
  "sideEffects": false,
  "dependencies": {
    "shell-escape": "^0.2.0",
    "iter-tools": "^6.2.6",
    "isomorphic-git": "^1.7.6",
    "fs-extra": "^10.0.0",
    "get-stdin": "^9.0.0",
    "yargs": "^17.0.0",
    "exec-once": "^0.1.8",
    "@tsfun/result": "^0.0.20",
    "@khai96x/srcinfo": "^0.0.5",
    "tslib": "^2.0.1",
    "@types/shell-escape": "^0.2.0",
    "@types/fs-extra": "^9.0.1",
    "@types/get-stdin": "^7.0.0",
    "@types/yargs": "^17.0.0",
    "@types/node": "^14.6.4"
  }
}
