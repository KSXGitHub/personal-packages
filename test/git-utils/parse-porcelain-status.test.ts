import { parsePorcelainStatus, parsePorcelainStatusValue } from '@khai96x/git-utils'

describe('parsePorcelainStatus', () => {
  it('success', () => {
    expect([
      '?? unstaged new file',
      ' D unstaged deleted file',
      ' M unstaged modified file',
      'A  staged new file',
      'D  staged deleted file',
      'M  staged modified file',
      'AD added and deleted',
      'AM added and modified',
      'MM modified twice',
    ].map(parsePorcelainStatus)).toMatchSnapshot()
  })

  it('separator is not a space', () => {
    expect(() => parsePorcelainStatus('??:new file')).toThrowErrorMatchingSnapshot()
  })
})

describe('parsePorcelainStatusValue', () => {
  it('success', () => {
    expect(
      ['??', ' M', ' D', 'A ', 'M ', 'D ', 'AD', 'AM', 'MD']
        .map(parsePorcelainStatusValue),
    ).toMatchSnapshot()
  })

  it('excessive characters', () => {
    expect(() => parsePorcelainStatusValue('???')).toThrowErrorMatchingSnapshot()
  })

  it('invalid staged character', () => {
    expect(() => parsePorcelainStatusValue('* ')).toThrowErrorMatchingSnapshot()
  })

  it('invalid unstaged character', () => {
    expect(() => parsePorcelainStatusValue(' A')).toThrowErrorMatchingSnapshot()
  })
})
