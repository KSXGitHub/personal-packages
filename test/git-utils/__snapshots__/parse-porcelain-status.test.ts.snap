// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parsePorcelainStatus separator is not a space 1`] = `"Third character of plain"`;

exports[`parsePorcelainStatus success 1`] = `
Array [
  Object {
    "path": "unstaged new file",
    "plain": "?? unstaged new file",
    "value": Object {
      "plain": "??",
      "staged": "?",
      "unstaged": "?",
    },
  },
  Object {
    "path": "unstaged deleted file",
    "plain": " D unstaged deleted file",
    "value": Object {
      "plain": " D",
      "staged": " ",
      "unstaged": "D",
    },
  },
  Object {
    "path": "unstaged modified file",
    "plain": " M unstaged modified file",
    "value": Object {
      "plain": " M",
      "staged": " ",
      "unstaged": "M",
    },
  },
  Object {
    "path": "staged new file",
    "plain": "A  staged new file",
    "value": Object {
      "plain": "A ",
      "staged": "A",
      "unstaged": " ",
    },
  },
  Object {
    "path": "staged deleted file",
    "plain": "D  staged deleted file",
    "value": Object {
      "plain": "D ",
      "staged": "D",
      "unstaged": " ",
    },
  },
  Object {
    "path": "staged modified file",
    "plain": "M  staged modified file",
    "value": Object {
      "plain": "M ",
      "staged": "M",
      "unstaged": " ",
    },
  },
  Object {
    "path": "added and deleted",
    "plain": "AD added and deleted",
    "value": Object {
      "plain": "AD",
      "staged": "A",
      "unstaged": "D",
    },
  },
  Object {
    "path": "added and modified",
    "plain": "AM added and modified",
    "value": Object {
      "plain": "AM",
      "staged": "A",
      "unstaged": "M",
    },
  },
  Object {
    "path": "modified twice",
    "plain": "MM modified twice",
    "value": Object {
      "plain": "MM",
      "staged": "M",
      "unstaged": "M",
    },
  },
]
`;

exports[`parsePorcelainStatusValue excessive characters 1`] = `"Excessive characters: ???"`;

exports[`parsePorcelainStatusValue invalid staged character 1`] = `"[?, A, D, M,  ] does not include *"`;

exports[`parsePorcelainStatusValue invalid unstaged character 1`] = `"[?, D, M,  ] does not include A"`;

exports[`parsePorcelainStatusValue success 1`] = `
Array [
  Object {
    "plain": "??",
    "staged": "?",
    "unstaged": "?",
  },
  Object {
    "plain": " M",
    "staged": " ",
    "unstaged": "M",
  },
  Object {
    "plain": " D",
    "staged": " ",
    "unstaged": "D",
  },
  Object {
    "plain": "A ",
    "staged": "A",
    "unstaged": " ",
  },
  Object {
    "plain": "M ",
    "staged": "M",
    "unstaged": " ",
  },
  Object {
    "plain": "D ",
    "staged": "D",
    "unstaged": " ",
  },
  Object {
    "plain": "AD",
    "staged": "A",
    "unstaged": "D",
  },
  Object {
    "plain": "AM",
    "staged": "A",
    "unstaged": "M",
  },
  Object {
    "plain": "MD",
    "staged": "M",
    "unstaged": "D",
  },
]
`;
